contracts/adapters/KycOnboarding.sol

```diff
@@ -0,0 +1,267 @@
+            "unitsPerChunk::invalid"
+        );
+        require(
+            maximumChunks * unitsPerChunk < type(uint88).max,
+            "potential overflow"
+        );
+
+        require(signerAddress != address(0x0), "signer address is nil!");
+
+        dao.setAddressConfiguration(SignerAddressConfig, signerAddress);
+        dao.setAddressConfiguration(FundTargetAddress, fundTargetAddress);
+        dao.setConfiguration(ChunkSize, chunkSize);
+        dao.setConfiguration(UnitsPerChunk, unitsPerChunk);
+        dao.setConfiguration(MaximumChunks, maximumChunks);
+        dao.setConfiguration(MaximumUnits, maxUnits);
+        dao.setConfiguration(MaxMembers, maxMembers);
+
+        BankExtension bank = BankExtension(
+            dao.getExtensionAddress(DaoHelper.BANK)
+        );
+        bank.registerPotentialNewInternalToken(DaoHelper.UNITS);
+    }
+
+    /**
+     * @notice Hashes the provided coupon as an ERC712 hash.
+     * @param dao is the DAO instance to be configured
+     * @param coupon is the coupon to hash
+     */
+    function hashCouponMessage(DaoRegistry dao, Coupon memory coupon)
+        public
+        view
+        returns (bytes32)
+    {
+        bytes32 message = keccak256(
+            abi.encode(COUPON_MESSAGE_TYPEHASH, coupon.kycedMember)
+        );
+
+        return hashMessage(dao, _chainId, address(this), message);
+    }
+
+    function _checkKycCoupon(
+        DaoRegistry dao,
+        address kycedMember,
+        bytes memory signature
+    ) internal view {
+        require(
+            ECDSA.recover(
+                hashCouponMessage(dao, Coupon(kycedMember)),
+                signature
+            ) == dao.getAddressConfiguration(SignerAddressConfig),
+            "invalid sig"
+        );
+    }
+
+    /**
+     * @notice Redeems a coupon to add a new member.
+     * @param dao is the DAO instance to be configured
+     * @param kycedMember is the address that this coupon authorized to become a new member
+     * @param signature is message signature for verification
+     */
+    function onboard(
+        DaoRegistry dao,
+        address kycedMember,
+        bytes memory signature
+    ) external payable reentrancyGuard(dao) {
+        require(!dao.isActiveMember(dao, kycedMember), "already member");
+        require(
+            dao.getNbMembers() < dao.getConfiguration(MaxMembers),
+            "the DAO is full"
+        );
+        _checkKycCoupon(dao, kycedMember, signature);
+        OnboardingDetails memory details = _checkData(dao);
+
+        BankExtension bank = BankExtension(
+            dao.getExtensionAddress(DaoHelper.BANK)
+        );
+        DaoHelper.potentialNewMember(kycedMember, dao, bank);
+        totalUnits[dao] += details.unitsRequested;
+        address payable multisigAddress = payable(
+            dao.getAddressConfiguration(FundTargetAddress)
+        );
+        if (multisigAddress == address(0x0)) {
```

Comment:
We may need a test case to onboard with ETH.

Reply:
this check is more whether we're using a multisig or not. But I agree we should test both cases